pkg/uploader/uploader.go

```diff
@@ -50,15 +53,49 @@ func (up *Uploader) Do(dstURL string) error {
 		return fmt.Errorf("invalid destination AMS URL ('%s'): %w", dstURL, err)
 	}
 	u.Path = path.Join(u.Path, "/sap/ams/v1/ams-instances/", up.AMSInstanceID, "/dcl-upload")
-	r, err := http.NewRequest(http.MethodPost, u.String(), buf)
+	resp, err := up.DoWithRetries(u.String(), body.Bytes(), maxRetries)
 	if err != nil {
-		return fmt.Errorf("could not create DCL upload request %w", err)
-	}
-	r.Header.Set("Content-Type", "application/gzip")
-	resp, err := up.Client.Do(r)
-	if err != nil {
-		return fmt.Errorf("DCL upload request unsuccessful: %w", err)
+		return fmt.Errorf("could not build upload request: %w", err)
 	}
 	defer resp.Body.Close()
 	return up.logResponse(resp)
```

Comment:
Body might be drained if after retry

Reply:
In which scenario do you think the body is drained? 
In my understanding, if there's a retry there are three cases:
* draining fails -> err != nil
* up.do() fails -> err != nil
* up.do() succeeds -> body is not drained